src/test/scala/com/microsoft/hyperspace/index/LogicalPlanSerDeTests.scala

```diff
@@ -73,6 +78,54 @@ class LogicalPlanSerDeTests extends SparkFunSuite with SparkInvolvedSuite {
     verifyPlanSerde(scanNode, "hadoopFsRelation.plan")
   }
 
+  private def verifyFileFormat(
+      fileFormat: FileFormat,
+      expectSerializationError: Boolean): Unit = {
+    val relation: HadoopFsRelation =
+      scanNode.relation.asInstanceOf[HadoopFsRelation].copy(fileFormat = fileFormat)(spark)
+    val updatedScanNode = scanNode.copy(relation = relation)
+
+    // This file format is serializable unless isSplittable is called on it. isSplittable api
+    // initializes some unserializable internal objects which break the serialization logic.
+    val kryoSerializer = new KryoSerializer(spark.sparkContext.getConf)
+    KryoSerDeUtils.serialize(kryoSerializer, fileFormat)
+
+    // Confirm that isSplittable makes serialization fail
+    fileFormat.isSplitable(spark, Map(), new Path("path"))
+    if (expectSerializationError) {
+      intercept[KryoException](KryoSerDeUtils.serialize(kryoSerializer, fileFormat))
+    } else {
+      KryoSerDeUtils.serialize(kryoSerializer, fileFormat)
+    }
+
+    // Now verify if Hyperspace serialization still works with this format
+    verifyPlanSerde(updatedScanNode, "hadoopFsRelation.plan")
+  }
+
+  test("Serde query with Hadoop file system parquet relation.") {
+    val parquetFormat = new ParquetFileFormat
+    // CSVFormat is unserializable due to internal unserializable members. Verify as below.
+    verifyFileFormat(parquetFormat, false)
```

Comment:
can we just do `verifyFileFormat(new ParquetFileFormat, false)`?

Reply:
thanks @imback82, done